cms/djangoapps/contentstore/rest_api/v2/views/downstreams.py

```diff
@@ -158,16 +162,74 @@ def get_paginated_response(self, data, *args, **kwargs):
         return response
 
 
-@view_auth_classes()
 class DownstreamListView(DeveloperErrorViewMixin, APIView):
     """
-    List all blocks which are linked to an upstream context, with optional filtering.
+    List all items (components and containers) wich are linked to an upstream context, with optional filtering.
+    """
+
+    def get(self, request: _AuthenticatedRequest):
+        """
+        Fetches publishable entity links for given course key
+        """
+        course_key_string = request.GET.get('course_id')
+        ready_to_sync = request.GET.get('ready_to_sync')
+        upstream_key = request.GET.get('upstream_key')
+        item_type = request.GET.get('item_type')
+        link_filter: dict[str, CourseKey | UsageKey | LibraryContainerLocator | bool] = {}
+        paginator = DownstreamListPaginator()
+
+        if course_key_string:
+            try:
+                link_filter["downstream_context_key"] = CourseKey.from_string(course_key_string)
+            except InvalidKeyError as exc:
+                raise ValidationError(detail=f"Malformed course key: {course_key_string}") from exc
```

Comment:
Shouldn't we check if the user has permission to view the course, if a course ID is specified?

And we should have some check if no course ID is specified - I don't think we should just allow any user to list all the links in every course. Maybe require that either an upstream or a downstream is set, and the user has permission to view whichever is set, unless they're a superuser.

Reply:
Updated: https://github.com/openedx/edx-platform/pull/37024/commits/d1ed1d1c7f5397afac3d97f98e6bf911d4968aac